src/wallet/wallet.h

```diff
@@ -475,7 +475,38 @@ class CWalletTx : public CMerkleTx
 };
 
 
-using CInputCoin = std::pair<const CWalletTx*, unsigned int>;
+class CInputCoin {
+public:
+    CInputCoin()

```

Comment:
> I don't think using optional would make the code more clear.

I think optional's useful. It saves work of having to build optionality into types you define, and more importantly it allows the same type to be optional in some contexts and required in others, which I think is exactly what you want here. std::optional is part of c++17 so eventually it will start becoming more commonly used.

Anyway, this was just a suggestion, so feel free to ignore it if you have a different preference.

> What about using CInputCoin* for coinLowestLarger instead?

Using a pointer was actually going to be my first suggestion, but I didn't see a way to make that work because coinLowestLarger is assigned from a local variable. But I think this would be an improvement if it could work.